components/search/BookingInfoSearch/SearchComponents/AdultsNumberInput.tsx

```diff
@@ -8,7 +8,7 @@ import {
 
 type AdultsNumberInputProps = {
   selectedNumber: number;
-  onChange: (number: number) => void;
+  onChange: (number: number | null) => void;
```

Comment:
Might be easier to constrain the value's to 1 or more. This can easily be done by setting the default value in the state too `1`.
Then have the `handleInputChange` set empty strings to 0, and update the value's state only if the new input is valid.

Reply:
On closer inspection, the previous implementation handles is mostly how I suggested so you could instead revert your changes on this file.